src/dinit.cc

```diff
@@ -696,6 +700,33 @@ int dinit_main(int argc, char **argv)
     close_control_socket();
     
     if (am_system_mgr) {
+        if (shutdown_type == shutdown_type_t::SOFTREBOOT) {
+            sync(); // Sync to minimise data loss if user elects to power off / hard reset
+
+            try {
+                char **argv_exec = new char*[argc + 1];
+                int j = 0;
+                for (int i = 0; i < argc; i++) {
+                    // strip out kernel cmdline: single
+                    if (strcmp(argv[i], "single") != 0) {
+                        argv_exec[j] = argv[i];
+                        ++j;
+                    }
+                }
+                argv_exec[j] = nullptr;
+
+                execv(dinit_exec, argv_exec);
+
+                free(argv_exec);
+            } catch (std::bad_alloc &exc) {
```

Comment:
From [Dinit's code style guide](https://github.com/davmac314/dinit/blob/master/doc/CODE-STYLE):
> 2. Curly braces - for classes and functions, the opening brace appears by itself on an empty line, as does the closing brace. In other cases the opening brace should appear at the end of the line (and not on a line by itself). The closing brace always appears on a line by itself.
```suggestion
            }
            catch (std::bad_alloc &exc) {
```